router.go

```diff
@@ -233,6 +250,16 @@ func DataResponse(data interface{}) APIResponse {
 	return APIResponse{Error: "", Status: "OK", Data: data}
 }
 
+// NewDataResponse creates new API data response using the resource
+func NewDataResponse(data interface{}) NewAPIResponse {
+	return NewAPIResponse{Error: nil, Status: "OK", Data: data}
+}
+
+// NewErrorResponse constructs error response from the API with error struct
+func NewErrorResponse(err *Error) NewAPIResponse {
```

Comment:
Sending the exact error message to the front end is not a good idea

Reply:
This does the same. The errors is predefined in pweb with appropriate messages.